File: API/Helpers/UserParams.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Helpers
{
  public enum OrderType{
    LastActive,
    Name,
    Age
  }
  public class UserParams : PaginationParams
  {
    public string CurrentUsername { get; set; }
    public string Predicate { get; set; }
    public string Gender { get; set; }
    public int MinAge { get; set; } = 1;
    public int MaxAge { get; set; } = 150;
    public OrderType OrderBy { get; set; } = OrderType.LastActive;
  }
}

File: client/src/app/_models/userParams.ts
import { BaseParams } from './baseParams';
import { User } from './user';

export const OrderType = {
  LastActive: 'lastActive',
  Name: 'name',
  Age: 'age'
} as const;

export class UserParams extends BaseParams {
  gender: string;
  minAge = 0;
  maxAge = 99;
  orderBy: string;
  predicate: string = 'liked';

  constructor(user: User) {
    super();
    this.gender = user.gender === 'female' ? 'male' : 'female';
    this.orderBy = OrderType.LastActive;
  }
}
